### PR TITLE
EOS-22900: CSM Web setup mini-provisioning interfaces

### PR DESCRIPTION
**Pull request by soniyamoholkar**
# Frontend

# Problem Statement

- _Overview: JIRA number/GitHub Issue added to PR: [Y/N]:_ EOS-22900
- _Describe the problem this patch intends to solve._ 
- CSM Web setup interfaces should be exposed for following mini-provisioning steps
- post_install
- prepare
- config
- init
- test
- reset
- cleanup (pre-factory)
- pre_upgrade
- post_upgrade

## Design
CSM web setup interfaces are exposed as defined in https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/85000402/Mini-Provisioner+ConfStore+Keys

## Coding

- _Coding conventions are followed and code is consistent. Yes/No?_ Yes
- _Confirm All CODACY errors are resolved. Yes/No?_ Yes

## Testing

- _Confirm that Test Cases are added (for both the cases, fix and feature). Yes/No?_ 
- _Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes/No?_
- _Confirm Testing was performed with installed RPM. Yes/No?_ Yes
- http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/cortx-management-portal/115/console
image
image

- _Confirm all the ut/regression/smoke test has been performed [Y/N]_

## Checklist

- _PR is self-reviewed? Yes/No?_ Yes
- _GitHub Issue is updated_
- _Jira is updated_
  -  _Check if the description is clear and explained._ 
    -  _Check Acceptance Criterion is defined._
      -  _All the tests performed should be mentioned before Resolving a JIRA._
        -  _Verification needs to be done before marked as Closed/Verified_
        - _Any interface change Yes/No?_
        - _Change notified to other gatekeepers: Yes/No?_
        - _Code reviews done and addressed: Yes/No?_
        - _Codacy issues reviewed and addressed: Yes/No?_
        - _Deployment test : Done/Not?_
        - _UT/smoke test: Done/Not?_
        - _Jira number added to PR: Yes/No?_
        - _Github merge done using UI: Yes/No?_
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_

### CODE REVIEW

File: web/conf/csm_web.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+class CSMWebSetupError(Exception):
+    """
+    Generic Exception with error code and output
+    """
+    def __init__(self, rc, message, *args):
+        """
+        Initializing CSMWebSetupError
+        """
+        self._rc = rc
+        self._desc = message % (args)
```
Comment (pawankumarsrivastava) [inline web/conf/csm_web.py:31]: Use fstring

File: web/conf/csm_web.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    CSM_WEB_DIST_ENV_FILE_PATH  = "/opt/seagate/cortx/csm/web/web-dist/.env"
+
+    def __init__(self, conf_url, **kwargs):
+        """
+        Initializing CSMWeb
+        """
+        Conf.init()
+        Conf.load(CSMWeb.CONSUMER_INDEX, conf_url)
+        Conf.load(self.ENV_INDEX, f"properties://{self.CSM_WEB_DIST_ENV_FILE_PATH }")
+        Log.init(service_name = "csm_web_setup", log_path = "/tmp",
```
Comment (pawankumarsrivastava) [inline web/conf/csm_web.py:61]: use /tmp/csm/setup_logs/

File: web/conf/csm_web_setup.py [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+    @staticmethod
+    def usage(prog: str):
+        """
+        Print usage instructions
+        """
+        sys.stderr.write(
+            f"usage: {prog} [-h] <cmd> --config <url> <args>...\n"
+            f"where:\n"
+            f"cmd   post_install, prepare, config, init, reset, test, pre_upgrade, post_upgrade\n"
```
Comment (pawankumarsrivastava) [inline web/conf/csm_web_setup.py:55]: add cleanup

File: web/conf/csm_web_setup.py [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        Override this method to add extended args
+        """
+        return 0
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """
+        Add Command args for parsing
+        """
+        parser1 = parser.add_parser(cls.name, help='setup %s' % name)
```
Comment (pawankumarsrivastava) [inline web/conf/csm_web_setup.py:86]: parser1 --> sub_parser

File: web/conf/csm_web_setup.py [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+        return rc
+
+def main(argv: dict):
+    try:
+        desc = "CORTX CSMWeb Setup command"
+        command = Cmd.get_command(desc, argv[1:])
+        rc = command.process()
+        if rc != 0:
+            raise ValueError(f"Failed to run {argv[1]}")
+        print(":PASS")
```
Comment (pawankumarsrivastava) [inline web/conf/csm_web_setup.py:259]: use sys.stdout.write

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: web/conf/csm_web.py (hunk 1 above)
Comment (soniyamoholkar) [reply web/conf/csm_web.py]: Done. Please check.

File: web/conf/csm_web.py (hunk 2 above)
Comment (soniyamoholkar) [reply web/conf/csm_web.py]: Done. Please check.

File: web/conf/csm_web_setup.py (hunk 3 above)
Comment (soniyamoholkar) [reply web/conf/csm_web_setup.py]: Done. Please check.

File: web/conf/csm_web_setup.py (hunk 4 above)
Comment (soniyamoholkar) [reply web/conf/csm_web_setup.py]: Done. Please check.

File: web/conf/csm_web_setup.py (hunk 5 above)
Comment (soniyamoholkar) [reply web/conf/csm_web_setup.py]: Done. Please check.

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by udayan-y21 — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**